rotkehlchen/chain/evm/decoding/zerox/decoder.py

```diff
@@ -41,37 +41,55 @@ def __init__(
             msg_aggregator: 'MessagesAggregator',
             router_address: ChecksumEvmAddress,
             flash_wallet_address: ChecksumEvmAddress,
+            settler_routers_addresses: list[ChecksumEvmAddress] | None = None,
     ) -> None:
         """router_address is the main point of contact with the 0x protocol.
         flash_wallet_address is a contract that can execute arbitrary calls from 0x router_address.
         Docs: https://0x.org/docs/introduction/0x-cheat-sheet#exchange-proxy-addresses"""
         super().__init__(evm_inquirer, base_tools, msg_aggregator)
         self.evm_txns = EvmTransactions(self.evm_inquirer, self.base.database)
         self.router_address = router_address
+        self.settler_routers_addresses = settler_routers_addresses
         self.flash_wallet_address = flash_wallet_address
+        self.router_addresses_set = {self.router_address}
+        if self.settler_routers_addresses:
+            self.router_addresses_set |= set(self.settler_routers_addresses)
 
     def _update_send_receive_fee_events(
             self,
             send_event: 'EvmEvent | None' = None,
             receive_event: 'EvmEvent | None' = None,
             fee_event: 'EvmEvent | None' = None,
+            router_address: ChecksumEvmAddress | None = None,
```

Comment:
this argument is not used then?

Reply:
It's used in line 218. This exists because if the swap is made via uniswap, the send_event will contain in address field the uniswap address, therefore necessary to change to the appropriate router address